### DISCUSSION TITLE
How to sink using S3 Extension in the example transfer-06 example pull

### DISCUSSION THREAD

**Discussion by PauloLory-ionos** [eclipse-edc/Samples#32] [Q&A]
I'm trying to do the [transfer-06-consumer-pull-http](https://github.com/eclipse-edc/Samples/tree/main/transfer/transfer-06-consumer-pull-http) example but instead of using the backend to get the file, I want to save the file into a repository like AWS S3, Azure Blob or IONOS S3.
The example as it is works fine but I changed the consumer to have also the IONOS S3 Extension and I would like to save the file into a S3 bucket after the pull process has finished.
I cannot seem to send the file that is already on the consumer side to the DP of the extension and perform the data sink in the cloud.
Some questions:

- 	Is there a way to somehow redirect the ending flow?
- 	Is there an API that I have to call after the “transferprocess”?
- 	Do I have to create a new DP being partially http-pull and partially S3 Sink?

Thanks

**Comment by Julian-Marco** [comment]
We're heavily interested in this too. As mentioned in this issue (https://github.com/eclipse-edc/Connector/issues/1903), the endpointOverride functionality doesn't make a lot of sense when pushing data. Currently we'd like to transfer data between two different MinIO instances by using the different S3 extensions, but the problem is that the provider connector always uses the endpoint that was given in its configuration file no matter what endpoint we set in the filetransfer.json. So pushing data via the S3 extensions only works when drastically changing the underlying code.

Could this be avoided when simply using the pull mechanism? I can't find an example for this, and the transfer-06-consumer-pull-http example doesn't seem to be suitable.
